packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js

```diff
@@ -240,34 +256,22 @@ export default class FormBuilderBlock extends React.Component {
     const memberType = this.getMemberTypeOf(value)
 
     return (
-      <FormBuilderInput
-        type={memberType}
-        level={0}
-        value={value}
-        onChange={this.handleChange}
-        onFocus={onFocus}
-        onBlur={onBlur}
-        focusPath={focusPath}
-        path={[{_key: value._key}]}
-      />
+      <div style={{minWidth: '30rem', padding: '1rem'}}>
+        <FormBuilderInput
+          type={memberType}
+          level={1}
+          value={value}
+          onChange={this.handleChange}
+          onFocus={onFocus}
+          onBlur={onBlur}
+          focusPath={focusPath}
+          path={[{_key: value._key}]}
+        />
+      </div>
     )
   }
   renderInput() {
-    const value = this.getValue()
-    const memberType = this.getMemberTypeOf(value)
-
-    const fieldsQty = ((memberType && memberType.fields) || []).length
-
-    let editModalLayout = get(this, 'props.type.options.editModal')
-
-    // Choose editModal based on number of fields
-    if (!editModalLayout) {
-      if (fieldsQty < 3) {
-        editModalLayout = 'popover'
-      } else {
-        editModalLayout = 'fullscreen'
-      }
-    }
+    const editModalLayout = get(this, 'props.type.options.editModal')
```

Comment:
It's better if you only get on the possible-null properties. In this case: `get(this.props.type.options, 'editModal')`

Reply:
Is that safe? `type` is not required in this component.

Reply:
We assume `type` is passed elsewhere in this component, so then it should be made required.